File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/domain/customer/valueobjects/FullName.java
package br.com.deltaglobalbank.customers.domain.customer.valueobjects;

public record FullName(String value) {
    public FullName {
        String fullName = value.trim();
        if (fullName.isBlank()) {
            throw new IllegalArgumentException("full_name_blank");
        }
        if (fullName.length() < 3 || fullName.length() > 255) {
            throw new IllegalArgumentException("full_name_invalid_length");
        }
        if (fullName.split("\\s+").length < 2) {
            throw new IllegalArgumentException("full_name_incomplete");
        }
        value = fullName;
    }
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/domain/customer/valueobjects/FullName.kt
package br.com.deltaglobalbank.customers.domain.customer.valueobjects

@JvmInline
value class FullName private constructor(val value: String) {
    companion object {
        operator fun invoke(raw: String): FullName {
            val fullName = raw.trim()
            require(fullName.isNotBlank()) { "full_name_blank" }
            require(fullName.length in 3..255) { "full_name_invalid_length" }
            require(fullName.split(Regex("\\s+")).size >= 2) { "full_name_incomplete" }
            return FullName(fullName)
        }
    }
}